### PR TITLE
Make Lifecycle ignore repeated stop requests

### PR DESCRIPTION
**Pull request by drcrallen**
Related to https://github.com/druid-io/druid/pull/1627

### CODE REVIEW

File: src/main/java/com/metamx/common/lifecycle/Lifecycle.java [hunk 1]
```diff
@@ -243,7 +250,10 @@ public void addMaybeStartHandler(Handler handler, Stage stage) throws Exception
   public void start() throws Exception
   {
     synchronized (handlers) {
-      started.set(true);
+      if (!started.compareAndSet(false, true)) {
```
Comment (gianm) [inline src/main/java/com/metamx/common/lifecycle/Lifecycle.java]: Should this be an error? I think unlike stop, there's no good reason for start to be called multiple times

Comment (drcrallen) [inline src/main/java/com/metamx/common/lifecycle/Lifecycle.java]: Maybe? Prior behavior did not throw an error, which is why I did not do so here.

Comment (gianm) [inline src/main/java/com/metamx/common/lifecycle/Lifecycle.java]: I feel like it makes sense to either not change the behavior, or to have it throw an error; silently skipping over double starts seems sketchy. At least the old behavior would probably blow up in some way and let us know something was being done wrong.

Comment (drcrallen) [inline src/main/java/com/metamx/common/lifecycle/Lifecycle.java]: Fixed

**Comment by himanshug** [comment]
:+1:

**Comment by drcrallen** [comment]
Bouncing for travis

**Comment by xvrl** [comment]
@drcrallen rebase on current master, it will help with travis

**Comment by drcrallen** [comment]
ding! winner

**Pull request merged**